src/modules/jcms_article/src/ArticleCrud.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace Drupal\jcms_article;
+
+use Drupal\Core\Entity\EntityTypeManager;
+use Drupal\jcms_article\Entity\ArticleVersions;
+use Drupal\node\Entity\Node;
+use Drupal\paragraphs\Entity\Paragraph;
+
+/**
+ * Class ArticleCrud
+ *
+ * @package Drupal\jcms_article
+ * @todo Look to share more code with \Drupal\jcms_article\Hooks\NodePreSave.
+ */
+class ArticleCrud {
+
+  /**
+   * @var \Drupal\Core\Entity\EntityTypeManager
+   */
+  protected $entityTypeManager;
+
+  /**
+   * ArticleCrud constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManager $entity_type_manager
+   */
+  public function __construct(EntityTypeManager $entity_type_manager) {
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * Helper method to decide whether to created, update or delete a node.
+   *
+   * @param \Drupal\jcms_article\Entity\ArticleVersions $articleVersions
+   *
+   * @return \Drupal\Core\Entity\EntityInterface|null
+   */
+  public function crudArticle(ArticleVersions $articleVersions) {
+    $node = NULL;
+    if ($nid = $this->getNodeIdByArticleId($articleVersions->getId())) {
+      if ($articleVersions->getAction() == ArticleVersions::DELETE) {
```

Comment:
According to @nlisgo if we try to access an article and it returns a 404 then it should be deleted. It may never happen but it may be useful to retain?